File: 2025/day1/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	fileObject, err := os.Open("input.txt")
	if err != nil {
		panic(err)
	}
	defer fileObject.Close()

	var lines []string
	scanner := bufio.NewScanner(fileObject)
	for scanner.Scan() {
		line := strings.TrimSpace(scanner.Text())
		lines = append(lines, line)
	}

	pointer := 50
	dialPoints := 0

	for _, line := range lines {
		dial := string(line[0])
		dial = strings.ToUpper(dial)
		number, _ := strconv.Atoi(line[1:])

		if dial == "L" {
			pointer = (pointer - number) % 100
			if pointer < 0 {
				pointer += 100
			}
		} else {
			pointer = (pointer + number) % 100
		}

		if pointer == 0 {
			dialPoints++
		}
	}

	fmt.Println(dialPoints)
}

File: 2025/day1/main.py
with open("input.txt", "r") as f:
    lines = [line.strip() for line in f]

position = 50
password = 0

for line in lines:
    direction = line[0]
    distance = int(line[1:])

    if direction == "L":
        position = (position - distance) % 100
    else:
        position = (position + distance) % 100

    if position == 0:
        password += 1

print(password)
